include/micm/solver/cuda_linear_solver.hpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2023 National Center for Atmospheric Research,
+//
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
```

Comment:
@mattldawson @K20shores In my opinion, the whole ODE solver (calculate Jacobian matrix, LU factorization, etc) could be named as a **linear solver**. What is really done here is the backward elimination based on the result of LU factorization. Calling it `linear_solver` is misleading and I would suggest renaming to `cuda_lu_solve` for GPU code and `lu_solve` for CPU code everywhere. Thoughts?

Reply:
Agreed. I will open a separate issue and assign to Qina later.